parsl/providers/local/local.py

```diff
@@ -76,19 +76,70 @@ def status(self, job_ids):
 
         logger.debug("Checking status of: {0}".format(job_ids))
         for job_id in self.resources:
-
-            retcode, stdout, stderr = self.channel.execute_wait('ps -p {} > /dev/null 2> /dev/null; echo "STATUS:$?" '.format(
-                self.resources[job_id]['remote_pid']), self.cmd_timeout)
-            for line in stdout.split('\n'):
-                if line.startswith("STATUS:"):
-                    status = line.split("STATUS:")[1].strip()
-                    if status == "0":
-                        self.resources[job_id]['status'] = JobStatus(JobState.RUNNING)
+            # This job dict should really be a class on its own
+            job_dict = self.resources[job_id]
+            if job_dict['status'] is not None and job_dict['status'].terminal:
```

Comment:
`if job_dict['status'] is not None ...` can be `if job_dict['status'] ...` ?

Reply:
If you think that's more readable...